engine/src/main/java/com/redhat/cloud/notifications/transformers/BaseTransformer.java

```diff
@@ -1,33 +1,52 @@
 package com.redhat.cloud.notifications.transformers;
 
 import com.redhat.cloud.notifications.ingress.Action;
+import com.redhat.cloud.notifications.templates.models.Environment;
 import io.vertx.core.json.JsonArray;
 import io.vertx.core.json.JsonObject;
 
 import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
 import java.util.Map;
 import java.util.stream.Collectors;
 
 @ApplicationScoped
 public class BaseTransformer {
 
-    public JsonObject transform(Action action) {
-        return toJsonObject(action);
-    }
+    @Inject
+    Environment environment;
 
-    protected JsonObject toJsonObject(Action action) {
+    /**
+     * Transforms the given action into a {@link JsonObject}.
+     * @param action the {@link Action} to transform.
+     * @return a {@link JsonObject} containing the given action.
+     */
+    public JsonObject toJsonObject(final Action action) {
         JsonObject message = new JsonObject();
-        message.put("bundle", action.getBundle());
+
+        message.put("account_id", action.getAccountId());
         message.put("application", action.getApplication());
+        message.put("bundle", action.getBundle());
+        message.put("context", JsonObject.mapFrom(action.getContext()));
+        // This class is often manually instantiated within some "TestHelpers" classes. In order to avoid hitting null
+        // pointer exceptions, this null check is required.
+        if (this.environment != null) {
+            message.put("environment_url", this.environment.url());
+        }
```

Comment:
Sorry, I did not anticipate that this would also be sent as part of email and webhook notifications which is not something we want. 99% of the changes in this PR still make sense, but now I'm not sure `BaseTransformer` is the best place to add `environment_url`.

`Environment` could be injected in `CamelTypeProcessor` and then this line
https://github.com/RedHatInsights/notifications-backend/blob/9f0526ddc6248dd334d0befc8e2cd99df4557c01/engine/src/main/java/com/redhat/cloud/notifications/processors/camel/CamelTypeProcessor.java#L158
would become
```
JsonObject payload = transformer.transform(event.getAction()); 
payload.put("environment_url", this.environment.url());
```
Or the signature of this method could have a boolean arg to specify if the environment URL should be included or not in the payload. And maybe an overload signature without that arg that would set it to `false` to avoid passing the arg at all calling points.
```
public JsonObject toJsonObject(final Action action) {
    return toJsonObject(action, false);
}

public JsonObject toJsonObject(final Action action, boolean includeEnvironmentUrl) {
[...]
        if (includeEnvironmentUrl && this.environment != null) {
            message.put("environment_url", this.environment.url());
        }
[...]
}
```

Reply:
Actually, even with what I just wrote, the URL will be sent to Camel which is again not something needed. It is only required for Slack notifications. Let me think a bit more about this... 😄

Reply:
https://github.com/RedHatInsights/notifications-backend/blob/9f0526ddc6248dd334d0befc8e2cd99df4557c01/engine/src/main/java/com/redhat/cloud/notifications/processors/camel/CamelTypeProcessor.java#L230

The `body` arg in that method is the payload that will be sent to RHOSE and transformed into a Slack notification. I think that arg is the one where `environment_url` should be added.

Reply:
Our Camel and RHOSE integrations are mixed in a single processor (CamelTypeProcessor) and that creates a lot of confusion... This is something I've wanted to fix for a while but I never got the time to take care of that.

Reply:
Splitting Camel & Splunk code might be something we should do in the short term - but i wonder if we should send across all messages the information about the environment in which the notification was generated (i.e. `environment: { type: prod, url: https://console.... }`

Reply:
FYI I created [RHCLOUD-22545](https://issues.redhat.com/browse/RHCLOUD-22545) for the split.

Reply:
All right, I've addressed the comments, and added a test for when we send events to Open Bridge. Now if we change anything from what we send to them, we should get a test failure for it.